modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,908 @@
+        array('t' => $titleProc)
+    );
+
+    if ($exists) {
+        showPublicationError('Submitted title already exists', 400);
+    }
+    $desc            = isset($_REQUEST['description'])
+        ? $_REQUEST['description'] : null;
+    $leadInvest      = isset($_REQUEST['leadInvestigator'])
+        ? $_REQUEST['leadInvestigator'] : null;
+    $leadInvestEmail = isset($_REQUEST['leadInvestigatorEmail'])
+        ? $_REQUEST['leadInvestigatorEmail'] : null;
+
+    // check if lead investigator already exists in collaborator table
+    // use ID if exists, else insert
+    $leadInvID = $db->pselectOne(
+        'SELECT PublicationCollaboratorID '.
+        'FROM publication_collaborator '.
+        'WHERE Name = :n AND Email = :e',
+        array(
+         'n' => $leadInvest,
+         'e' => $leadInvestEmail,
+        )
+    );
+    if (empty($leadInvID)) {
+        $db->insert(
+            'publication_collaborator',
+            array(
+             'Name'  => $leadInvest,
+             'Email' => $leadInvestEmail,
+            )
+        );
+
+        $leadInvID = $db->getLastInsertId();
+    }
+    if (!isset($desc, $leadInvest, $leadInvestEmail)) {
+        showPublicationError('A mandatory field is missing!', 400);
+    }
+    // INSERT INTO publication ...
+    $uid   = $user->getId();
+    $today = date('Y-m-d');
+    // insert the titleRaw to avoid double escaping
+    $fields = array(
+               'UserID'             => $uid,
+               'Title'              => $titleRaw,
+               'Description'        => $desc,
+               'LeadInvestigatorID' => $leadInvID,
+               'DateProposed'       => $today,
+              );
+
+    $db->insert('publication', $fields);
+    $pubID = $db->getLastInsertId();
+
+    try {
+        // process files
+        processFiles($pubID);
+        // INSERT INTO publication_collaborator
+        insertCollaborators($pubID);
+        // INSERT INTO publication_users_edit_perm_rel
+        insertEditors($pubID);
+        // INSERT INTO publication_keyword
+        insertKeywords($pubID);
+        // INSERT INTO publication_parameter_type_rel
+        insertVOIs($pubID);
+    } catch (Exception $e) {
+        cleanup($pubID);
+        showPublicationError($e->getMessage(), $e->getCode());
+    }
+
+    notify($pubID, 'submission');
+}
+
+/**
+ * Function for storing files and inserting file meta data into database
+ *
+ * @param int $pubID ID of Publication in publication table
+ *
+ * @return void
+ */
+function processFiles($pubID) : void
+{
+    if (empty($_FILES)) {
+        return;
+    }
+    $db     = \Database::singleton();
+    $config = \NDB_Config::singleton();
+
+    $publicationPath = $config->getSetting('publication_uploads');
+
+    if (!isset($publicationPath)) {
+        throw new LorisException(
+            "Error! Publication path is not set in Loris Settings!",
+            500
+        );
+    }
+
+    if (!file_exists($publicationPath)) {
```

Comment:
```suggestion
    if (!is_dir($publicationPath)) {
```
Can also make sure that it is truly a directory and not a file by mistake.